extensions/dacpac/src/wizard/pages/deployActionPage.ts

```diff
@@ -169,7 +169,8 @@ export class DeployActionPage extends DacFxConfigPage {
 	private setDefaultScriptFilePath(): void {
 		let now = new Date();
 		let datetime = now.getFullYear() + '-' + (now.getMonth() + 1) + '-' + now.getDate() + '-' + now.getHours() + '-' + now.getMinutes();
-		this.fileTextBox.value = path.join(os.homedir(), this.model.database + '_UpgradeDACScript_' + datetime + '.sql');
+		let rootPath = vscode.workspace.rootPath ? vscode.workspace.rootPath : os.homedir();
```

Comment:
This line of code is repeated a lot. Can you put it in a method to centralize the logic?

Reply:
I moved the one in the dacpac extension to a method in the base page class